stream_alert/alert_processor/output_base.py

```diff
@@ -0,0 +1,319 @@
+
+        # Get the decrypted credential json from kms and load into dict
+        # This could be None if the kms decryption fails, so check it
+        decrypted_creds = self._kms_decrypt(enc_creds)
+        if not decrypted_creds:
+            return
+
+        creds_dict = json.loads(decrypted_creds)
+
+        # Add any of the hard-coded default output props to this dict (ie: url)
+        defaults = self._get_default_properties()
+        if defaults:
+            creds_dict.update(defaults)
+
+        return creds_dict
+
+    def _format_s3_bucket(self, suffix):
+        """Format the s3 bucket by combining the stored qualifier with a suffix
+
+        Args:
+            suffix [string]: Suffix for an s3 bucket
+
+        Returns:
+            [string] The combined prefix and suffix
+        """
+        return '.'.join([self.s3_prefix, suffix])
+
+    def _get_creds_from_s3(self, cred_location, descriptor):
+        """Pull the encrypted credential blob for this service and destination from s3
+
+        Args:
+            cred_location [string]: The tmp path on disk to to store the encrypted blob
+            descriptor [string]: Service destination (ie: slack channel, pd integration)
+
+        Returns:
+            [boolean] True if download of creds from s3 was a success
+        """
+        try:
+            client = boto3.client('s3', region_name=self.region)
+            with open(cred_location, 'wb') as cred_output:
+                client.download_fileobj(self.get_secrets_bucket_name(),
+                                        self.output_cred_name(descriptor),
+                                        cred_output)
+
+            return True
+        except ClientError as err:
+            LOGGER.error('credentials for %s could not be downloaded from S3: %s',
+                         self.output_cred_name(descriptor),
+                         err.response)
+
+    def _kms_decrypt(self, data):
+        """Decrypt data with AWS KMS.
+
+        Args:
+            data [string]: An encrypted ciphertext data blob
+
+        Returns:
+            [string] Decrypted json string
+        """
+        try:
+            client = boto3.client('kms', region_name=self.region)
+            response = client.decrypt(CiphertextBlob=data)
+            return response['Plaintext']
+        except ClientError as err:
+            LOGGER.error('an error occurred during credentials decryption: %s', err.response)
+
+    def _log_status(self, success):
+        """Log the status of sending the alerts
+
+        Args:
+            success [boolean]: Indicates if the dispatching of alerts was successful
+        """
+        if success:
+            LOGGER.info('successfully sent alert to %s', self.__service__)
+        else:
+            LOGGER.error('failed to send alert to %s', self.__service__)
+
+    @staticmethod
+    def _format_prefix(s3_prefix):
+        """Return a bucket prefix that has been properly formatted
```

Comment:
There is `_format_prefix` and `_format_s3_bucket`. These functions are kind of confusing to me - what are they formatting and why? What does it mean to be `properly formatted`? Maybe explain or give examples of input/output.

Each of those functions are only used once, so it may actually be conceptually simpler to just do them inline instead of breaking them out as separate private static functions